workers/tests/unittests/test_api_request_models.py

```diff
@@ -19,59 +20,126 @@ def setUp(self):
+        request_models.BaseDatabase(
             dataset="regions", feature_id="3", fid_field="ogc_fid"
         )
 
     def test_dataset_invalid(self):
         with self.assertRaises(ValueError):
-            request_models.BasePOSTModel(dataset="foo", feature_id="3")
+            request_models.BaseDatabase(dataset="foo", feature_id="3")
         with self.assertRaises(ValueError):
-            request_models.BaseGETModel(dataset="foo", feature_id="3")
+            request_models.BaseDatabase(dataset="foo", feature_id="3")
 
-    def test_invalid_set_of_arguments(self):
-        with self.assertRaises(ValueError):
-            request_models.BasePOSTModel(
-                bpolys=self.bpolys, dataset="regions", feature_id="3"
-            )
-        with self.assertRaises(ValueError):
-            request_models.BasePOSTModel(dataset="regions")
-        with self.assertRaises(ValueError):
-            request_models.BaseGETModel(dataset="regions")
-        with self.assertRaises(ValueError):
-            request_models.BasePOSTModel(feature_id="3")
-        with self.assertRaises(ValueError):
-            request_models.BaseGETModel(feature_id="3")
-        with self.assertRaises(ValueError):
-            request_models.BaseIndicatorModel(name="GhsPopComparisonBuildings")
-        with self.assertRaises(ValueError):
-            request_models.BaseIndicatorModel(layer_name="building_count")
-
-    def test_valid_indicator_layer_combination(self):
-        request_models.BaseIndicatorModel(
+    def test_valid_layer(self):
+        request_models.BaseIndicator(
             name="GhsPopComparisonBuildings",
             layerName="building_count",
         )
 
+    def test_invalid_layer(self):
+        with self.assertRaises(ValueError):
+            request_models.BaseIndicator(
+                name="GhsPopComparisonBuildings",
+                layerName="foo",
+            )
+
     def test_invalid_indicator_layer_combination(self):
         with self.assertRaises(ValueError):
-            request_models.BaseIndicatorModel(
+            request_models.BaseIndicator(
                 name="GhsPopComparisonBuildings",
                 layerName="amenities",
             )
+
+    def test_indicator_database(self):
+        request_models.IndicatorDatabase(
+            name="GhsPopComparisonBuildings",
+            layerName="building_count",
+            dataset="regions",
+            featureId="3",
+        )
+        request_models.IndicatorDatabase(
+            name="GhsPopComparisonBuildings",
+            layerName="building_count",
+            dataset="regions",
+            featureId="Heidelberg",
+            fidField="name",
+        )
+
+    def test_indicator_bpolys(self):
+        request_models.IndicatorBpolys(
+            name="GhsPopComparisonBuildings",
+            layerName="building_count",
+            bpolys=self.bpolys,
+        )
+
+    def test_invalid_set_of_arguments(self):
```

Comment:
maybe we need tests for that test :stuck_out_tongue_winking_eye:

Reply:
Would probably be good :D